### PR TITLE
PICARD-851: Correctly update parent items on save

### PR DESCRIPTION
**Pull request by samj1912**
When multiple files were linked to the same track and the changes to
the items were saved, the parent album icon didn't update to reflect
the change.

This was because of a buggy item.save() function which led to an update
call to the update() method of files. The files on update did not recu-
rsively call their parent items to be updated too.

### CODE REVIEW

**Comment by mwiencek** [comment]
Looks good, I think, though I haven't touched this code in years.

File: picard/ui/itemviews.py [hunk 1]
```diff
@@ -728,25 +728,26 @@ def update(self, update_album=True):
...
-                for i in xrange(newnum - 1, oldnum - 1, -1):
-                    item = FileItem(track.linked_files[i], False)
-                    item.update()
-                    items.append(item)
-                self.addChildren(items)
+            if update_files:
+                oldnum = self.childCount()
+                newnum = track.num_linked_files
+                if oldnum > newnum:  # remove old items
+                    for i in xrange(oldnum - newnum):
```
Comment (zas) [inline picard/ui/itemviews.py]: `xrange()` doesn't exist in python 3, perhaps better to use `range()`.

Comment (samj1912) [inline picard/ui/itemviews.py]: I just shifted that bit from the existing code. Will make the changes :)

File: picard/ui/itemviews.py [hunk 2]
```diff
@@ -755,13 +756,14 @@ def update(self, update_album=True):
             self.setBackground(i, bgcolor)
         if self.isSelected():
             TreeItem.window.update_selection()
+
```
Comment (mwiencek) [inline picard/ui/itemviews.py]: Sorry to nitpick, but can you remove the unrelated whitespace change?
